src/test/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProviderTest.java

```diff
@@ -0,0 +1,262 @@
+                        .name("normal")
+                        .build(),
+                Queue.builder()
+                        .name("partition1")
+                        .build(),
+                Queue.builder()
+                        .name("partition2")
+                        .build());
+        final List<Queue> partitionNames = slurmQueueProvider.listQueues();
+        Assertions.assertEquals(partitionsName, partitionNames);
+    }
+
+    @Test
+    public void listPartitionsWithFilterSuccessful() {
+        final CommandResult listResult = CommandResult.builder()
+                .stdOut(validStdout)
+                .stdErr(Collections.emptyList())
+                .build();
+        doReturn(listResult).when(mockCmdExecutor).execute(Mockito.any());
+
+        final List<Queue> filtredPartition =
+                List.of(Queue.builder()
+                        .name(NEW_PARTITION_NAME)
+                        .hostList(nodeListOneNode)
+                        .slots(new SlotsDescription(1,
+                                slotDescriptionOneNode))
+                        .allowedUserGroups(groupsList)
+                        .build());
+        final QueueFilter filter = QueueFilter.builder().queues(List.of("normal")).build();
+        final List<Queue> partitionsFiltrationResult = slurmQueueProvider.listQueues(filter);
+        Assertions.assertEquals(filtredPartition, partitionsFiltrationResult);
+    }
+
+    @Test
+    public void deletePartitionNotFound() {
+        final CommandResult deletionFailedResult = CommandResult.builder()
+                .stdOut(Collections.emptyList())
+                .stdErr(List.of("delete_partition PartitionName=deletedPartition: Invalid partition name "
+                        + "specified"))
+                .build();
+        doReturn(deletionFailedResult).when(mockCmdExecutor).execute(Mockito.any());
+
+        final QueueVO incorrectDeletionData = QueueVO.builder()
+                .name(DELETED_PARTITION_NAME)
+                .build();
+        final GridEngineException gridExceptionUpdate = Assertions.assertThrows(GridEngineException.class,
+                () -> slurmQueueProvider.updateQueue(incorrectDeletionData));
+        Assertions.assertNotNull(gridExceptionUpdate.getMessage());
+    }
+
+    @Test
+    public void deletePartitionSuccessful() {
+        final CommandResult deletePartitionResult = CommandResult.builder()
+                .stdErr(Collections.emptyList())
+                .build();
+
+        doReturn(deletePartitionResult).when(mockCmdExecutor).execute(Mockito.any());
+
+        final Queue deletedPartition = Queue.builder()
+                .name(DELETED_PARTITION_NAME)
+                .build();
+        final Queue deletedPartitionResult = slurmQueueProvider.deleteQueues(DELETED_PARTITION_NAME);
+        Assertions.assertEquals(deletedPartition, deletedPartitionResult);
+    }
+
+    static Stream<Arguments> incorrectDataForPartitionCreateAndUpdate() {
```

Comment:
please place this method next the appropriate parameterized test

Reply:
done